### ISSUE TITLE
[Feature] System RW support

### ISSUE THREAD

**Issue by misaka19465** [tiann/KernelSU#767]
### Is your feature request related to a problem? Please describe.

Viper4Android FX and some other apps doesn't support ksu and could only be actived by modifying system partition.

### Describe the solution you'd like.

Some old apps (like Viper4Android FX) and some old-fashioned people may still need the function of writeable system and other read-only partitions. So I'm considering to develop a feature, when it's enabled, modules would be unmounted next boot, and a empty img would be mounted as rw. Then when you disable it, the img would be converted to tar. 

### Describe alternatives you've considered.

Create a empty img and mount as upper dir.

### Additional context

I'm already working on it, and I wonder is it a reasonable feature and how to solve something like oplus kernels.

**Comment by tiann** [comment]
KernelSU is born for new devices, compatibility with old devices and old modules is not our main purpose.

We had planned to develop this feature but later abandoned it, because we found that the vast majority of people who need this feature have no idea what they are doing.
If you want to add this feature, we won't object either. But we hope this feature will be like this:

1. The manager will not provide any UI for it. Users will need to manually create system_rw.img or other files under /data/adb/ksu or some other directory, to make /system(/product)(/vendor) writable when overlayfs mounts.
2. The documentation will mark this as an experimental feature, and suggest users use modules instead wherever possible.
3. This feature is not guaranteed to work. Any issues related to system rw will be closed immediately.
In summary, we won't stop you from adding it, but will not provide official support. It should be an advanced experimental option without UI. We hope to minimize negative impact by warning users and marking it clearly as unstable and dangerous.

You can create a draft PR, so i will close this issue.

**Comment by dilumpallege123-collab** [comment]
Viper4android
